src/coreneuron/io/reports/report_handler.cpp

```diff
@@ -6,13 +6,106 @@
 # =============================================================================
 */
 
+#include <sstream>
 #include "report_handler.hpp"
 #include "coreneuron/io/nrnsection_mapping.hpp"
 #include "coreneuron/mechanism/mech_mapping.hpp"
 #include "coreneuron/utils/utils.hpp"
 
 namespace coreneuron {
 
+SectionType check_section_type(SectionType st) {
+    if (st == SectionType::All)
+        return SectionType::All;
+    if (st == SectionType::Cell)
+        return SectionType::Soma;
+    if (st == SectionType::Soma)
+        return SectionType::Soma;
+    if (st == SectionType::All)
+        return SectionType::Axon;
+    if (st == SectionType::Dendrite)
+        return SectionType::Dendrite;
+    if (st == SectionType::Apical)
+        return SectionType::Apical;
+
+    std::cerr << "[Error] Invalid SectionType enum value: " << to_string(st) << "\n";
+    nrn_abort(1);
+}
+
+/**
+ * @brief Retrieves a pointer to a variable associated with a mechanism or state variable.
+ *
+ * This function returns a pointer to the memory location of a variable specified by
+ * its mechanism name and variable name for a given segment in the neuron thread.
+ *
+ * Special cases:
+ * - If `mech_name` is "i_membrane", returns the pointer to the membrane current (`nrn_sav_rhs`)
+ *   at the specified `segment_id`.
+ * - If `mech_name` is "v", returns the pointer to the membrane voltage (`_actual_v`)
+ *   at the specified `segment_id`.
+ *
+ * For other mechanisms:
+ * - Retrieves the `Memb_list` corresponding to `mech_id` from the neuron thread.
+ * - Uses `segment_id_2_node_id` as a cache mapping segment IDs to node indices.
+ *   If the cache is empty, it populates it by iterating over all nodes in `ml`.
+ * - Returns the pointer to the variable obtained via `get_var_location_from_var_name`.
+ *
+ * @param nt Reference to the NrnThread containing mechanism data.
+ * @param mech_id Mechanism type ID.
+ * @param var_name Name of the variable to retrieve within the mechanism.
+ * @param mech_name Name of the mechanism (e.g., "i_membrane", "v", or other mechanisms).
+ * @param segment_id ID of the segment for which to retrieve the variable.
+ * @param segment_id_2_node_id Cache mapping segment IDs to node indices; populated if empty.
+ * @return Pointer to the requested variable's memory location, or nullptr if mechanism not found.
+ */
+double* get_var(const NrnThread& nt,
```

Comment:
I'm curious whether the function name could be more...descriptive I guess? I do see that it _does_ "get the variable", but in a somewhat specific context. Is this part of the public API or not? If not, I'm happy with leaving it as-is.

Reply:
it is used only in this translation unit. I will leave it like that for now